eclipse-collections/src/main/java/org/eclipse/collections/impl/lazy/CompositeRichIterable.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (c) 2016 Goldman Sachs.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v. 1.0 which accompany this distribution.
+ * The Eclipse Public License is available at http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ */
+
+package org.eclipse.collections.impl.lazy;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+import net.jcip.annotations.Immutable;
+import org.eclipse.collections.api.RichIterable;
+import org.eclipse.collections.api.block.predicate.Predicate;
+import org.eclipse.collections.api.block.procedure.Procedure;
+import org.eclipse.collections.api.block.procedure.Procedure2;
+import org.eclipse.collections.api.block.procedure.primitive.ObjectIntProcedure;
+import org.eclipse.collections.api.list.MutableList;
+import org.eclipse.collections.impl.Counter;
+import org.eclipse.collections.impl.EmptyIterator;
+import org.eclipse.collections.impl.list.mutable.FastList;
+import org.eclipse.collections.impl.utility.Iterate;
+
+@Immutable
+public final class CompositeRichIterable<E>
+        extends AbstractLazyIterable<E>
+{
+    private final MutableList<RichIterable<E>> iterables;
+
+    private CompositeRichIterable(MutableList<RichIterable<E>> newIterables)
+    {
+        this.iterables = newIterables;
+    }
+
+    public CompositeRichIterable()
+    {
+        this(FastList.newList());
+    }
+
+    public static <T> CompositeRichIterable<T> with(RichIterable<T>... iterables)
+    {
+        return new CompositeRichIterable<>(FastList.newListWith(iterables));
+    }
+
+    @Override
+    public void each(Procedure<? super E> procedure)
+    {
+        this.iterables.each(iterable -> Iterate.forEach(iterable, procedure));
```

Comment:
Isn't `iterable` a `RichIterable`? then you can do:

`this.iterables.each(iterable -> iterable.each(procedure));`